src/compute/src/render/reduce.rs

```diff
@@ -208,37 +208,50 @@ where
             // If we have no aggregations or just a single type of reduction, we
             // can go ahead and render them directly.
             ReducePlan::Distinct => {
-                let (arranged_output, errs) = self.dispatch_build_distinct(collection, mfp_after);
+                let (arranged_output, errs) = {
+                    let collection = collection.map(|(k, v)| {
+                        assert!(v.is_empty());
+                        (k, Row::default())
+                    });
```

Comment:
I think this block can just go away? Not sure about the types just from reading the PR, but if `v` is a row, then we assert it to be empty and replace it by a default row. I'd argue that any plan leaving us with a non-empty `v` here is incorrect, so we don't need to assert the fact it is. (It's not something we enforce elsewhere, and we'd probably fail later with an equally confusing error about too many columns.)

Reply:
Works for me!